src/app/clusters/ota-requestor/OTARequestor.h

```diff
@@ -78,17 +85,86 @@ class OTARequestor : public OTARequestorInterface
     };
 
+    // TODO: the application should define this, along with initializing the BDXDownloader
+
+    // This class is purely for delivering messages and sending outgoing messages to/from the BDXDownloader.
+    class BDXMessenger : public chip::BDXDownloader::MessagingDelegate, public chip::Messaging::ExchangeDelegate
+    {
+    public:
+        CHIP_ERROR SendMessage(const chip::bdx::TransferSession::OutputEvent & event) override
+        {
+            ChipLogDetail(SoftwareUpdate, "BDX::SendMessage");
+            VerifyOrReturnError(mExchangeCtx != nullptr, CHIP_ERROR_INCORRECT_STATE);
+
+            chip::Messaging::SendFlags sendFlags;
+            if (event.msgTypeData.HasMessageType(chip::bdx::MessageType::ReceiveInit))
+            {
+                sendFlags.Set(chip::Messaging::SendMessageFlags::kFromInitiator);
+            }
+            if (!event.msgTypeData.HasMessageType(chip::bdx::MessageType::BlockAckEOF) &&
+                !event.msgTypeData.HasMessageType(chip::Protocols::SecureChannel::MsgType::StatusReport))
+            {
+                sendFlags.Set(chip::Messaging::SendMessageFlags::kExpectResponse);
+            }
+            ReturnErrorOnFailure(mExchangeCtx->SendMessage(event.msgTypeData.ProtocolId, event.msgTypeData.MessageType,
+                                                           event.MsgData.Retain(), sendFlags));
+            return CHIP_NO_ERROR;
+        }
+
+        CHIP_ERROR OnMessageReceived(chip::Messaging::ExchangeContext * ec, const chip::PayloadHeader & payloadHeader,
+                                     chip::System::PacketBufferHandle && payload) override
+        {
+            if (mDownloader == nullptr)
+            {
+                ChipLogError(BDX, "BDXDownloader instance is null, can't pass message");
+                return CHIP_NO_ERROR;
+            }
+            else
+            {
+                mDownloader->OnMessageReceived(payloadHeader, payload.Retain());
+            }
+
+            // For a receiver using BDX Protocol, all received messages will require a response except for a StatusReport
+            if (!payloadHeader.HasMessageType(chip::Protocols::SecureChannel::MsgType::StatusReport))
+            {
+                ec->WillSendMessage();
+            }
+            return CHIP_NO_ERROR;
+        }
+
+        void OnResponseTimeout(chip::Messaging::ExchangeContext * ec) override
+        {
+            ChipLogError(BDX, "exchange timed out");
+            if (mDownloader != nullptr)
+            {
+                mDownloader->OnDownloadTimeout();
+            }
+        }
+
+        void Init(chip::BDXDownloader * downloader, chip::Messaging::ExchangeContext * ec)
+        {
+            mExchangeCtx = ec;
+            mDownloader  = downloader;
+        }
+
+    private:
+        chip::Messaging::ExchangeContext * mExchangeCtx;
+        chip::BDXDownloader * mDownloader;
+    };
+
     // Variables
+    // TODO: align on variable naming standard
```

Comment:
Are you referring to using `m` for member variables or something else?

Reply:
yes exactly that

Reply:
I've addressed this in my PR